runmariadb.sh

```diff
@@ -12,6 +24,16 @@ if [ ! -d "${DATADIR}/mysql" ]; then
     crudini --set "$MARIADB_CONF_FILE" mysqld innodb_buffer_pool_size 5M
     crudini --set "$MARIADB_CONF_FILE" mysqld innodb_log_buffer_size 512K
 
+    # Config MariaDB to enable TLS
+    if [ -f "$MARIADB_CERT_FILE" ]
+    then
+      mkdir -p "${DATADIR}/certs/"
+      cp "/certs/mariadb" "${DATADIR}/certs/" -r
```

Comment:
What is the point of this? It has the effect of copying the private key to a volume where it may get persisted to disk.

Reply:
Thanks for your review. The reason for this copying to the database directory is that the certificate and key needs to be owned by user `mysql`. Otherwise, the database cannot read these files. I am not sure if  it is a good idea if we change the ownership of files in a mounted volume. Besides, TLS can only be enabled when the volume is mounted, so I am not sure if we gain any security benefits when not storing the key in the container. What is your idea about this, @zaneb ?

Reply:
Your clarification makes sense that we can gain secuirty benefit when not storing the key file on disk. 
The reason why we needed the key file being owned by `mysql` is that its default permission is 600 (only owner can read and write).
```sh 
stat -c '%a' /certs/mariadb/tls.key
600
```
What if we change the permission of the key file to 604 before mount it to the container? WDYT @zaneb @maelk ?

Reply:
Thanks for the clarification. I agree the answer is to mount it with the right permissions, rather than try to change it in the container.
How are you mounting the secret now? This should just work by default when mounted from a Secret. e.g. I checked with the existing mounts used for basic auth and the default mode is 644:

```
sh-4.4# ls -Hl /auth/ironic/password
-rw-r--r--. 1 root root 16 Oct 22 14:29 /auth/ironic/password
```

Reply:
From metal3-dev-env, we have two options to mount the key to the container:
- Using secret: I didn't check if the key has the permission 644 by default. I agree with you that by default the file mounted by a secret has permission 644, but the private key file (generated by Openssl, for example) is explicitly set to 600 when it is created. I am not sure by being mounted to a secret, this permission can be changed to 644.

Reply:
For a Secret the permissions on the file generated by openssl are irrelevant, since you're going to copy the contents of the file into the Secret. What matters is [the permissions for the mounted Secret](https://kubernetes.io/docs/concepts/configuration/secret/#secret-files-permissions) which can be modified but default to 644 which is what we want anyway.

If you use a bind mount then I agree that you will need to manually set the appropriate mode on the file. However, nobody would ever use this method in a real deployment, and for dev-env purposes we can just put it in the script.

Reply:
The bind mount is used when Ironic run standalone outside of the K8s cluster. 
Anyway, I have updated the PR. Please take a look. Thanks